src/ansys/mapdl/core/component.py

```diff
@@ -0,0 +1,449 @@
+from ansys.mapdl.core.mapdl import _MapdlCore
+
+if TYPE_CHECKING:  # pragma: no cover
+    import logging
+
+ENTITIES_TYP = Literal[
+    "NODE", "NODES", "ELEM", "ELEMS", "ELEMENTS", "VOLU", "AREA", "LINE", "KP"
+]
+
+VALID_ENTITIES = list(get_args(ENTITIES_TYP))
+
+SELECTOR_FUNCTION = [
+    "NSEL",
+    "NSEL",
+    "ESEL",
+    "ESEL",
+    "ESEL",
+    "VSEL",
+    "ASEL",
+    "LSEL",
+    "KSEL",
+]
+
+ENTITIES_MAPPING = {
+    entity.upper(): func for entity, func in zip(VALID_ENTITIES, SELECTOR_FUNCTION)
+}
+
+
+ITEMS_VALUES = Optional[Union[str, int, list[int], NDArray[Any]]]
+UNDERLYING_DICT = dict[str, ITEMS_VALUES]
+
+warning_entity = (
+    "Assuming a {default_entity} selection.\n"
+    "It is recommended you use the following notation to avoid this warning:\n"
+    ">>> mapdl.components['{key}'] = '{default_entity}' {value}\n"
+    "Alternatively, you disable this warning using:\n"
+    ">>> mapdl.components.default_entity_warning=False"
+)
+
+
+def _check_valid_pyobj_to_entities(
+    items: Union[tuple[int, ...], list[int], NDArray[Any]]
+):
+    """Check whether the python objects can be converted to entities.
+    At the moment, only list and numpy arrays of ints are allowed.
+    """
+
+    if not all([isinstance(each, (int, np.integer)) for each in items]):
+        raise ValueError("Only integers are allowed for component definition.")
+
+
+class Component(tuple):
+    """Component object
+
+    Object which contain the definition of a component.
+
+    Parameters
+    ----------
+    type_ : str
+        The entity type. For instance "NODES", "KP", "VOLU", etc
+
+    items_ : None, str, int, list[int], np.array[int]]
+        Item ids contained in the component.
+
+    Examples
+    --------
+    To create a component object
+
+    >>> mycomp = Component("NODES", [1, 2, 3])
+    >>>  mycomp
+    Component(type='NODES', items=(1, 2, 3))
+
+    To index the content of the component object:
+
+    >>> mycomp[0]
+    1
+    >>> mycomp[1]
+    2
+
+    To access the type of the entities:
+
+    >>> mycomp.type
+    "NODES"
+
+    Convert to list:
+
+    >>> list(mycomp)
+    [1, 2, 3]
+    """
+
+    def __new__(
+        cls,
+        type_: ENTITIES_TYP,
+        items_: tuple[str, Union[tuple[int], list[int], NDArray[np.int_]]],
+    ):
+        if not isinstance(type_, str) or type_.upper() not in VALID_ENTITIES:
+            raise ValueError(
+                f"The value '{type_}' is not allowed for 'type' definition."
+            )
+        obj = super().__new__(cls, items_)
+        obj._type: ENTITIES_TYP = type_
+
+        return obj
+
+    def __str__(self):
+        tup_str = super().__str__()
+        return f"Component(type='{self._type}', items={tup_str})"
+
+    def __repr__(self) -> str:
+        tup_str = super().__repr__()
+        return f"Component(type='{self._type}', items={tup_str})"
+
+    @property
+    def type(self) -> str:
```

Comment:
Here you can explicitly say that you are returning an object of type `ENTITIES_TYP`
```suggestion
    def type(self) -> ENTITIES_TYP:
```